extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java

```diff
@@ -146,6 +154,8 @@ public CompletableResultCode shutdown() {
 
```

Comment:
in the `CompletableResultCode shutdown()` should the `client.close();` join the current thread? (similar for the corresponding shutdown of the HttpExporter)

Reply:
Good catch!

Reply:
We should be returning a `CompletableResultCode` that completes when the client closes. Do you want to open a separate PR for that?

Reply:
yes, happy to do that for both exporters. Does something as simple as replacing `client.close()` to `client.close().result()` be sufficient here or do we want any special error handling if there is any thrown exception?
Also, not very familiar with the Vertx library, in case there is a more preferred way for waiting the result to get computed.

Reply:
I would do something like this:

```java
        var result = new CompletableResultCode();
        client.close()
                .onSuccess(
                    new Handler<>() {
                        @Override
                        public void handle(Void event) {
                            result.succeed();
                        }
                    })
                .onFailure(new Handler<>() {
                    @Override
                    public void handle(Throwable event) {
                        result.fail();
                    }
                });
        return result;
```

Reply:
thanks for the hint @geoand , looking at the docs, does this code block until `close()` is executed?

This could possibly be related to the closed connection errors we are seeing as @gsmet mentioned. if the shutdown is called before flushing the pending requests it could result in this error

Reply:
>  does this code block until close() is executed

Does what block?

Reply:
nevermind, I got confused with the blocking/non-blocking semantics here. What you suggested makes sense now that I looked deeper into the code